### PR TITLE
use local typescript instead of global typescript executable

### PR DESCRIPTION
**Pull request by momozor**
I think it will be better if we use explicit local typescript executable that can be installed locally, so the user doesn't have to install a system wide typescript which more than often involves administrator/root permission requirement.

This patch should work fine with all Build & Test command as well.

### CODE REVIEW

File: ref_impl/package.json [hunk 1]
```diff
@@ -11,14 +11,15 @@
         "url": "https://github.com/Microsoft/BosqueLanguage"
     },
     "dependencies": {
-        "@types/node": "~10.12.18"
+        "@types/node": "~10.12.18",
+        "typescript": "^3.4.3"
```
Comment (Hongarc) [inline ref_impl/package.json]: You should add it to `devDependencies` instead of `dependencies`

Comment (momozor) [reply ref_impl/package.json]: Hi @Hongarc. Thanks for the review.

I think if I put it in `devDependencies` instead, people won't be able to build the software in `ref_impl/bin/` which contains the interpreter, compiler to build/run their code if they do just run install without dev dependencies install switch.

What do you think?

Comment (Hongarc) [reply ref_impl/package.json]: > people won't be able to build the software builds
If they are the user, they don't need to build `bin` file.
We publish `bin` file in the release.

Comment (momozor) [reply ref_impl/package.json]: If that is the case, then sure. Thanks.

**Pull request closed without merging**